### PR TITLE
Remove unnecessary raw pointer in __rust_start_panic arg

### PR DESCRIPTION
**Pull request by bjorn3**
It is no longer necessary as __rust_start_panic switched to the Rust abi.

### CODE REVIEW

File: library/panic_unwind/src/lib.rs [hunk 1]
```diff
@@ -98,7 +98,7 @@ pub unsafe extern "C" fn __rust_panic_cleanup(payload: *mut u8) -> *mut (dyn Any
 // Entry point for raising an exception, just delegates to the platform-specific
 // implementation.
 #[rustc_std_internal_symbol]
-pub unsafe fn __rust_start_panic(payload: *mut &mut dyn BoxMeUp) -> u32 {
+pub unsafe fn __rust_start_panic(payload: &mut dyn BoxMeUp) -> u32 {
     let payload = Box::from_raw((*payload).take_box());
```
Comment (RalfJung) [inline library/panic_unwind/src/lib.rs:102]: The `*` here can be removed now, right?

File: library/panic_abort/src/android.rs [hunk 2]
```diff
@@ -15,7 +15,7 @@ type SetAbortMessageType = unsafe extern "C" fn(*const libc::c_char) -> ();
 //
 // Weakly resolve the symbol for android_set_abort_message. This function is only available
 // for API >= 21.
-pub(crate) unsafe fn android_set_abort_message(payload: *mut &mut dyn BoxMeUp) {
+pub(crate) unsafe fn android_set_abort_message(payload: &mut dyn BoxMeUp) {
```
Comment (RalfJung) [inline library/panic_abort/src/android.rs:18]: There's another `*` below that can be removed I think, line 26.

File: library/panic_unwind/src/lib.rs (hunk 1 above)
Comment (bjorn3) [reply library/panic_unwind/src/lib.rs]: Indeed. Fixed.

**Comment by Aaron1011** [comment]
I believe this will require a change to the Miri panic implementation as well.

**Comment by bjorn3** [comment]
Miri uses `miri_start_panic` as unwind entrypoint, which is called by panic_unwind's `__rust_start_panic`, so no miri changes needed afaict.

**Comment by Aaron1011** [comment]
For reference, my previous comment was "I think this will require changes to Miri".  I deleted it shortly after posting it, as I realized it was incorrect :)

File: library/std/src/panicking.rs [hunk 3]
```diff
@@ -51,7 +51,7 @@ extern "Rust" {
     /// `payload` is passed through another layer of raw pointers as `&mut dyn Trait` is not
```
Comment (BlackHoleFox) [inline library/std/src/panicking.rs:51]: This comment seems stale since its just `&mut dyn Trait` now. The `improper_ctypes` lint also seems unneeded now.

Comment (bjorn3) [reply library/std/src/panicking.rs]: Updated.

**Comment by ChrisDenton** [comment]
This looks like a nice improvement. Maybe squish the commits though?

**Comment by bjorn3** [comment]
Rebased and squashed.

**Pull request merged**